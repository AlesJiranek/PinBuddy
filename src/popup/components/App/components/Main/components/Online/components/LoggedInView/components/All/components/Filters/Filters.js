import React, { Component } from "react";
import PropTypes from "prop-types";

import { Input, Button, Checkbox } from "theme";
import "./Filters.css";

class Filters extends Component {
  state = {
    showFilters: false
  };

  render() {
    const {
      privatePost,
      publicPost,
      unread,
      untagged,
      updateKeyword,
      updatePrivatePost,
      updatePublicPost,
      updateUnread,
      updateUntagged
    } = this.props;

    return (
      <div className="filters">
        <div className="filters__top">
          <div className="filters__input">
            <Input
              id="filter"
              label={chrome.i18n.getMessage("popupFiltersSearchLabel")}
              placeholder={chrome.i18n.getMessage(
                "popupFiltersSearchPlaceholder"
              )}
              onChange={updateKeyword}
              autoFocus
            />
          </div>
          <div className="filters__button">
            <Button
              t={chrome.i18n.getMessage("popupFiltersFilterButton")}
              onClick={this.toggleFilters}
            />
          </div>
        </div>
        <div
          className={
            this.state.showFilters
              ? "filters__bottom filters__bottom--active"
              : "filters__bottom"
          }
        >
          <div className="filters__option">
            <Checkbox
              id="privatePost"
              label={chrome.i18n.getMessage("popupFiltersCheckboxPrivate")}
              checked={privatePost}
              onChange={updatePrivatePost}
            />
          </div>
          <div className="filters__option">
            <Checkbox
              id="publicPost"
              label={chrome.i18n.getMessage("popupFiltersCheckboxPublic")}
              checked={publicPost}
              onChange={updatePublicPost}
            />
          </div>
          <div className="filters__option">
            <Checkbox
              id="unread"
              label={chrome.i18n.getMessage("popupFiltersCheckboxUnread")}
              checked={unread}
              onChange={updateUnread}
            />
          </div>
          <div className="filters__option">
            <Checkbox
              id="untagged"
              label={chrome.i18n.getMessage("popupFiltersCheckboxUntagged")}
              checked={untagged}
              onChange={updateUntagged}
            />
          </div>
        </div>
      </div>
    );
  }

  toggleFilters = () => {
    this.setState(state => {
      return {
        showFilters: !state.showFilters
      };
    });
  };
}

Filters.propTypes = {
  privatePost: PropTypes.bool.isRequired,
  publicPost: PropTypes.bool.isRequired,
  unread: PropTypes.bool.isRequired,
  untagged: PropTypes.bool.isRequired,
  updateKeyword: PropTypes.func.isRequired,
  updatePrivatePost: PropTypes.func.isRequired,
  updatePublicPost: PropTypes.func.isRequired,
  updateUnread: PropTypes.func.isRequired,
  updateUntagged: PropTypes.func.isRequired
};

export default Filters;
