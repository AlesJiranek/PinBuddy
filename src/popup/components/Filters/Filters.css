.filters__top {
  display: flex;
  align-items: flex-end;
}

.filters__input {
  width: 100%;
}

.filters__button {
  margin-left: var(--lh);
}

.filters__bottom {
  display: none;
}

.filters__bottom--active {
  display: flex;
}

.all__filter {
  margin-right: var(--lh);
  text-overflow: ellipsis;
  white-space: nowrap;
  animation-name: slideIn;
  animation-duration: var(--animation-duration);
  animation-fill-mode: backwards;
}

.all__filter:nth-child(2) {
  animation-delay: 50ms;
}

.all__filter:nth-child(3) {
  animation-delay: 100ms;
}

.all__filter:nth-child(4) {
  animation-delay: 150ms;
}
