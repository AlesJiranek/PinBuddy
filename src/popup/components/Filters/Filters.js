import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Input, Button, Checkbox } from 'theme';
import './Filters.css';

class Filters extends Component {
  state = {
    showFilters: false,
  }

  render() {
    const {
      toRead,
      privatePosts,
      updateKeyword,
      updateToRead,
      updatePrivatePosts,
    } = this.props;

    return (
      <div className="filters">
        <div className="filters__top">
          <div className="filters__input">
            <Input
              id="filter"
              label="Search"
              placeholder="Search term"
              onChange={updateKeyword}
            />
          </div>
          <div className="filters__button">
            <Button
              t="Filter"
              onClick={this.toggleFilters}
            />
          </div>
        </div>
        <div className={this.state.showFilters ? 'filters__bottom filters__bottom--active' : 'filters__bottom'}>
          <div className="all__filter">
            <Checkbox
              id="toRead"
              label="Uread"
              checked={toRead}
              onChange={updateToRead}
            />
          </div>
          <div className="all__filter">
            <Checkbox
              id="privatePosts"
              label="Private"
              checked={privatePosts}
              onChange={updatePrivatePosts}
            />
          </div>
        </div>

      </div>
    );
  }

  toggleFilters = () => {
    this.setState(state => {
      return {
        showFilters: !state.showFilters,
      };
    });
  }
}

Filters.propTypes = {
  toRead: PropTypes.bool.isRequired,
  privatePosts: PropTypes.bool.isRequired,
  updateKeyword: PropTypes.func.isRequired,
  updateToRead: PropTypes.func.isRequired,
  updatePrivatePosts: PropTypes.func.isRequired,
};

export default Filters;

