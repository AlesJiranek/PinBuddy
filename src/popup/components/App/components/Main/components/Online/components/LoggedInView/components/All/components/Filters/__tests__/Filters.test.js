import React from 'react';
import { shallow } from 'enzyme';

import Filters from '../Filters';

describe('<Filters />', () => {

  const defaultProps = {
    privatePost: true,
    publicPost: true,
    unread: true,
    untagged: true,
    updateKeyword: jest.fn(),
    updatePrivatePost: jest.fn(),
    updatePublicPost: jest.fn(),
    updateUnread: jest.fn(),
    updateUntagged: jest.fn(),
  };

  it('should render correctly', () => {
    const tree = shallow(
      <Filters
        {...defaultProps}
      />
    );
    expect(tree).toMatchSnapshot();
  });

});
