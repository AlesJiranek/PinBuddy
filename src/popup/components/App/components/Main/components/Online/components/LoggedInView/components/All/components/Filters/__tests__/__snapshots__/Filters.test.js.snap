// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filters /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Filters
    privatePost={true}
    publicPost={true}
    unread={true}
    untagged={true}
    updateKeyword={[MockFunction]}
    updatePrivatePost={[MockFunction]}
    updatePublicPost={[MockFunction]}
    updateUnread={[MockFunction]}
    updateUntagged={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="filters__top"
        >
          <div
            className="filters__input"
          >
            <ForwardRef
              autoCapitalize={null}
              autoComplete={null}
              autoCorrect={null}
              autoFocus={true}
              id="filter"
              invalidMessage={null}
              label="popupFiltersSearchLabel"
              onChange={[MockFunction]}
              pattern={null}
              placeholder="popupFiltersSearchPlaceholder"
              spellCheck={null}
              type="text"
            />
          </div>
          <div
            className="filters__button"
          >
            <Button
              disabled={false}
              onClick={[Function]}
              secondary={false}
              t="popupFiltersFilterButton"
              title={null}
              type="button"
            />
          </div>
        </div>,
        <div
          className="filters__bottom"
        >
          <div
            className="filters__option"
          >
            <Checkbox
              checked={true}
              id="privatePost"
              label="popupFiltersCheckboxPrivate"
              onChange={[MockFunction]}
            />
          </div>
          <div
            className="filters__option"
          >
            <Checkbox
              checked={true}
              id="publicPost"
              label="popupFiltersCheckboxPublic"
              onChange={[MockFunction]}
            />
          </div>
          <div
            className="filters__option"
          >
            <Checkbox
              checked={true}
              id="unread"
              label="popupFiltersCheckboxUnread"
              onChange={[MockFunction]}
            />
          </div>
          <div
            className="filters__option"
          >
            <Checkbox
              checked={true}
              id="untagged"
              label="popupFiltersCheckboxUntagged"
              onChange={[MockFunction]}
            />
          </div>
        </div>,
      ],
      "className": "filters",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="filters__input"
            >
              <ForwardRef
                autoCapitalize={null}
                autoComplete={null}
                autoCorrect={null}
                autoFocus={true}
                id="filter"
                invalidMessage={null}
                label="popupFiltersSearchLabel"
                onChange={[MockFunction]}
                pattern={null}
                placeholder="popupFiltersSearchPlaceholder"
                spellCheck={null}
                type="text"
              />
            </div>,
            <div
              className="filters__button"
            >
              <Button
                disabled={false}
                onClick={[Function]}
                secondary={false}
                t="popupFiltersFilterButton"
                title={null}
                type="button"
              />
            </div>,
          ],
          "className": "filters__top",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ForwardRef
                autoCapitalize={null}
                autoComplete={null}
                autoCorrect={null}
                autoFocus={true}
                id="filter"
                invalidMessage={null}
                label="popupFiltersSearchLabel"
                onChange={[MockFunction]}
                pattern={null}
                placeholder="popupFiltersSearchPlaceholder"
                spellCheck={null}
                type="text"
              />,
              "className": "filters__input",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "autoCapitalize": null,
                "autoComplete": null,
                "autoCorrect": null,
                "autoFocus": true,
                "id": "filter",
                "invalidMessage": null,
                "label": "popupFiltersSearchLabel",
                "onChange": [MockFunction],
                "pattern": null,
                "placeholder": "popupFiltersSearchPlaceholder",
                "spellCheck": null,
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "autoCapitalize": null,
                  "autoComplete": null,
                  "autoCorrect": null,
                  "autoFocus": null,
                  "invalidMessage": null,
                  "pattern": null,
                  "spellCheck": null,
                  "type": "text",
                },
                "propTypes": Object {
                  "autoCapitalize": [Function],
                  "autoComplete": [Function],
                  "autoCorrect": [Function],
                  "autoFocus": [Function],
                  "id": [Function],
                  "invalidMessage": [Function],
                  "label": [Function],
                  "onChange": [Function],
                  "pattern": [Function],
                  "spellCheck": [Function],
                  "type": [Function],
                },
                "render": [Function],
              },
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Button
                disabled={false}
                onClick={[Function]}
                secondary={false}
                t="popupFiltersFilterButton"
                title={null}
                type="button"
              />,
              "className": "filters__button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "disabled": false,
                "onClick": [Function],
                "secondary": false,
                "t": "popupFiltersFilterButton",
                "title": null,
                "type": "button",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="filters__option"
            >
              <Checkbox
                checked={true}
                id="privatePost"
                label="popupFiltersCheckboxPrivate"
                onChange={[MockFunction]}
              />
            </div>,
            <div
              className="filters__option"
            >
              <Checkbox
                checked={true}
                id="publicPost"
                label="popupFiltersCheckboxPublic"
                onChange={[MockFunction]}
              />
            </div>,
            <div
              className="filters__option"
            >
              <Checkbox
                checked={true}
                id="unread"
                label="popupFiltersCheckboxUnread"
                onChange={[MockFunction]}
              />
            </div>,
            <div
              className="filters__option"
            >
              <Checkbox
                checked={true}
                id="untagged"
                label="popupFiltersCheckboxUntagged"
                onChange={[MockFunction]}
              />
            </div>,
          ],
          "className": "filters__bottom",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Checkbox
                checked={true}
                id="privatePost"
                label="popupFiltersCheckboxPrivate"
                onChange={[MockFunction]}
              />,
              "className": "filters__option",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "checked": true,
                "id": "privatePost",
                "label": "popupFiltersCheckboxPrivate",
                "onChange": [MockFunction],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Checkbox
                checked={true}
                id="publicPost"
                label="popupFiltersCheckboxPublic"
                onChange={[MockFunction]}
              />,
              "className": "filters__option",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "checked": true,
                "id": "publicPost",
                "label": "popupFiltersCheckboxPublic",
                "onChange": [MockFunction],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Checkbox
                checked={true}
                id="unread"
                label="popupFiltersCheckboxUnread"
                onChange={[MockFunction]}
              />,
              "className": "filters__option",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "checked": true,
                "id": "unread",
                "label": "popupFiltersCheckboxUnread",
                "onChange": [MockFunction],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Checkbox
                checked={true}
                id="untagged"
                label="popupFiltersCheckboxUntagged"
                onChange={[MockFunction]}
              />,
              "className": "filters__option",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "checked": true,
                "id": "untagged",
                "label": "popupFiltersCheckboxUntagged",
                "onChange": [MockFunction],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="filters__top"
          >
            <div
              className="filters__input"
            >
              <ForwardRef
                autoCapitalize={null}
                autoComplete={null}
                autoCorrect={null}
                autoFocus={true}
                id="filter"
                invalidMessage={null}
                label="popupFiltersSearchLabel"
                onChange={[MockFunction]}
                pattern={null}
                placeholder="popupFiltersSearchPlaceholder"
                spellCheck={null}
                type="text"
              />
            </div>
            <div
              className="filters__button"
            >
              <Button
                disabled={false}
                onClick={[Function]}
                secondary={false}
                t="popupFiltersFilterButton"
                title={null}
                type="button"
              />
            </div>
          </div>,
          <div
            className="filters__bottom"
          >
            <div
              className="filters__option"
            >
              <Checkbox
                checked={true}
                id="privatePost"
                label="popupFiltersCheckboxPrivate"
                onChange={[MockFunction]}
              />
            </div>
            <div
              className="filters__option"
            >
              <Checkbox
                checked={true}
                id="publicPost"
                label="popupFiltersCheckboxPublic"
                onChange={[MockFunction]}
              />
            </div>
            <div
              className="filters__option"
            >
              <Checkbox
                checked={true}
                id="unread"
                label="popupFiltersCheckboxUnread"
                onChange={[MockFunction]}
              />
            </div>
            <div
              className="filters__option"
            >
              <Checkbox
                checked={true}
                id="untagged"
                label="popupFiltersCheckboxUntagged"
                onChange={[MockFunction]}
              />
            </div>
          </div>,
        ],
        "className": "filters",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="filters__input"
              >
                <ForwardRef
                  autoCapitalize={null}
                  autoComplete={null}
                  autoCorrect={null}
                  autoFocus={true}
                  id="filter"
                  invalidMessage={null}
                  label="popupFiltersSearchLabel"
                  onChange={[MockFunction]}
                  pattern={null}
                  placeholder="popupFiltersSearchPlaceholder"
                  spellCheck={null}
                  type="text"
                />
              </div>,
              <div
                className="filters__button"
              >
                <Button
                  disabled={false}
                  onClick={[Function]}
                  secondary={false}
                  t="popupFiltersFilterButton"
                  title={null}
                  type="button"
                />
              </div>,
            ],
            "className": "filters__top",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <ForwardRef
                  autoCapitalize={null}
                  autoComplete={null}
                  autoCorrect={null}
                  autoFocus={true}
                  id="filter"
                  invalidMessage={null}
                  label="popupFiltersSearchLabel"
                  onChange={[MockFunction]}
                  pattern={null}
                  placeholder="popupFiltersSearchPlaceholder"
                  spellCheck={null}
                  type="text"
                />,
                "className": "filters__input",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "autoCapitalize": null,
                  "autoComplete": null,
                  "autoCorrect": null,
                  "autoFocus": true,
                  "id": "filter",
                  "invalidMessage": null,
                  "label": "popupFiltersSearchLabel",
                  "onChange": [MockFunction],
                  "pattern": null,
                  "placeholder": "popupFiltersSearchPlaceholder",
                  "spellCheck": null,
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": Object {
                    "autoCapitalize": null,
                    "autoComplete": null,
                    "autoCorrect": null,
                    "autoFocus": null,
                    "invalidMessage": null,
                    "pattern": null,
                    "spellCheck": null,
                    "type": "text",
                  },
                  "propTypes": Object {
                    "autoCapitalize": [Function],
                    "autoComplete": [Function],
                    "autoCorrect": [Function],
                    "autoFocus": [Function],
                    "id": [Function],
                    "invalidMessage": [Function],
                    "label": [Function],
                    "onChange": [Function],
                    "pattern": [Function],
                    "spellCheck": [Function],
                    "type": [Function],
                  },
                  "render": [Function],
                },
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Button
                  disabled={false}
                  onClick={[Function]}
                  secondary={false}
                  t="popupFiltersFilterButton"
                  title={null}
                  type="button"
                />,
                "className": "filters__button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "disabled": false,
                  "onClick": [Function],
                  "secondary": false,
                  "t": "popupFiltersFilterButton",
                  "title": null,
                  "type": "button",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="filters__option"
              >
                <Checkbox
                  checked={true}
                  id="privatePost"
                  label="popupFiltersCheckboxPrivate"
                  onChange={[MockFunction]}
                />
              </div>,
              <div
                className="filters__option"
              >
                <Checkbox
                  checked={true}
                  id="publicPost"
                  label="popupFiltersCheckboxPublic"
                  onChange={[MockFunction]}
                />
              </div>,
              <div
                className="filters__option"
              >
                <Checkbox
                  checked={true}
                  id="unread"
                  label="popupFiltersCheckboxUnread"
                  onChange={[MockFunction]}
                />
              </div>,
              <div
                className="filters__option"
              >
                <Checkbox
                  checked={true}
                  id="untagged"
                  label="popupFiltersCheckboxUntagged"
                  onChange={[MockFunction]}
                />
              </div>,
            ],
            "className": "filters__bottom",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Checkbox
                  checked={true}
                  id="privatePost"
                  label="popupFiltersCheckboxPrivate"
                  onChange={[MockFunction]}
                />,
                "className": "filters__option",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "checked": true,
                  "id": "privatePost",
                  "label": "popupFiltersCheckboxPrivate",
                  "onChange": [MockFunction],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Checkbox
                  checked={true}
                  id="publicPost"
                  label="popupFiltersCheckboxPublic"
                  onChange={[MockFunction]}
                />,
                "className": "filters__option",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "checked": true,
                  "id": "publicPost",
                  "label": "popupFiltersCheckboxPublic",
                  "onChange": [MockFunction],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Checkbox
                  checked={true}
                  id="unread"
                  label="popupFiltersCheckboxUnread"
                  onChange={[MockFunction]}
                />,
                "className": "filters__option",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "checked": true,
                  "id": "unread",
                  "label": "popupFiltersCheckboxUnread",
                  "onChange": [MockFunction],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Checkbox
                  checked={true}
                  id="untagged"
                  label="popupFiltersCheckboxUntagged"
                  onChange={[MockFunction]}
                />,
                "className": "filters__option",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "checked": true,
                  "id": "untagged",
                  "label": "popupFiltersCheckboxUntagged",
                  "onChange": [MockFunction],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
